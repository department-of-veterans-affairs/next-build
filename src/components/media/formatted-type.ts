// import { MediaImage } from '@/components/mediaDocument/formatted-type'
import { PublishedParagraph } from '@/types/formatted/publishedEntity'

export type Media = PublishedParagraph & {
  type: 'paragraph--media'
  // image: MediaImage
  image: {
    url: string
    alt: string
    title: string | null
  }
  allowClicksOnThisImage: boolean
}
