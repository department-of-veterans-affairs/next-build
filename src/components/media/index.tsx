import Image from '@/components/image'
import { DrupalFile } from 'next-drupal'
import { absoluteURL } from '@/utils/utils'

interface ImageProps {
  url: string
  width?: string
  height?: string
  alt?: string
  title?: string
  styles?: object
}

interface MediaProps {
  media: DrupalFile
  imageStyle: string
}

export function formatImage(file: DrupalFile): ImageProps {
  return {
    url: absoluteURL(file?.image?.uri?.url),
    styles: file?.image?.links,
    alt: file?.image?.resourceIdObjMeta?.alt,
    title: file?.image?.resourceIdObjMeta?.title,
    width: file?.image?.resourceIdObjMeta?.width,
    height: file?.image?.resourceIdObjMeta?.height,
  }
}

export const MediaImage = ({ media, imageStyle }: MediaProps) => {
  if (!media) return null
  const { styles, url, alt, title, width, height } = formatImage(media)

  if (!imageStyle) imageStyle = 'full_content_width'

  const imageStyles = {
    url: styles[imageStyle]?.href,
    height: styles[imageStyle]?.meta?.height,
    width: styles[imageStyle]?.meta?.width,
  }

  return (
    <Image
      src={imageStyles ? imageStyles.url : url}
      alt={alt}
      title={title}
      width={imageStyles ? imageStyles.width : width}
      height={imageStyles ? imageStyles.height : height}
    />
  )
}
