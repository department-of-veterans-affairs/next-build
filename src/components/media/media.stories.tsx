import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { MediaImageComponent } from './index'

const data = {
  media: {
    type: 'media--image',
    url: 'https://www.example.com/image.jpg',
    id: '1',
    imageStyle: 'string',
    image: {
      uri: {
        url: 'https://www.example.com/image.jpg',
      },
      resourceIdObjMeta: {
        alt: 'pension',
        height: '100',
        title: 'title',
        width: '100',
      },
      links: {
        '1_1_square_medium_thumbnail': {
          href: 'https://www.example.com/image.jpg',
          meta: {
            height: '100',
            width: '100',
          },
        },
      },
    },
  },
}

export default {
  title: 'Components/Media',
  component: MediaImageComponent,
} as ComponentMeta<typeof MediaImageComponent>

const Template: ComponentStory<typeof MediaImageComponent> = (args) => (
  <MediaImageComponent {...args} />
)

export const Primary = Template.bind({})
Primary.args = {
  image: data.media,
  imageStyle: 'full_content_width',
}
