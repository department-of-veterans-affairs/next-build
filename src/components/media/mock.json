{
  "type": "paragraph--media",
  "id": "bc5d91c0-ae80-4b24-a942-e2c3688bbe8d",
  "drupal_internal__id": 183009,
  "drupal_internal__revision_id": 1895600,
  "langcode": "en",
  "status": true,
  "created": "2025-10-16T18:26:44+00:00",
  "parent_id": "43882",
  "parent_type": "node",
  "parent_field_name": "field_content_block",
  "behavior_settings": [],
  "default_langcode": true,
  "revision_translation_affected": true,
  "breadcrumbs": [],
  "content_translation_source": "und",
  "content_translation_outdated": false,
  "content_translation_changed": "2025-10-16T18:26:44+00:00",
  "field_allow_clicks_on_this_image": true,
  "field_markup": null,
  "links": {
    "self": {
      "href": "http://va-gov-cms.ddev.site/jsonapi/paragraph/media/bc5d91c0-ae80-4b24-a942-e2c3688bbe8d?resourceVersion=id%3A1895600"
    }
  },
  "resourceIdObjMeta": {
    "target_revision_id": 1895600,
    "drupal_internal__target_id": 183009
  },
  "paragraph_type": {
    "type": "paragraphs_type--paragraphs_type",
    "id": "71445226-8425-4b28-a78e-a6a6cce821a6",
    "resourceIdObjMeta": {
      "drupal_internal__target_id": "media"
    }
  },
  "field_media": {
    "type": "media--image",
    "id": "9b657b3a-530d-4fb7-9c98-72c1ce81043d",
    "drupal_internal__mid": 46787,
    "drupal_internal__vid": 48870,
    "status": true,
    "name": "my-image.png",
    "created": "2025-10-16T18:27:37+00:00",
    "changed": "2025-10-16T18:27:37+00:00",
    "breadcrumbs": [
      {
        "uri": "http://va-gov-cms.ddev.site/",
        "title": "Home",
        "options": []
      },
      {
        "uri": "http://va-gov-cms.ddev.site/media/image/46787",
        "title": "My-image.png",
        "options": []
      }
    ],
    "path": {
      "alias": "/media/image/46787",
      "pid": 134643,
      "langcode": "en"
    },
    "field_description": null,
    "links": {
      "self": {
        "href": "http://va-gov-cms.ddev.site/jsonapi/media/image/9b657b3a-530d-4fb7-9c98-72c1ce81043d?resourceVersion=id%3A48870"
      }
    },
    "resourceIdObjMeta": {
      "drupal_internal__target_id": 46787
    },
    "bundle": {
      "type": "media_type--media_type",
      "id": "92f64a34-81e5-4094-82a9-4fe61d7f2227",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": "image"
      }
    },
    "uid": {
      "type": "user--user",
      "id": "7c46746a-7527-4403-abca-1c115643037a",
      "resourceIdObjMeta": {
        "drupal_internal__target_id": 5409
      }
    },
    "thumbnail": {
      "type": "file--file",
      "id": "44ac8309-f23e-4286-85ce-f4958314e1bb",
      "drupal_internal__fid": 53380,
      "langcode": "en",
      "filename": "my-image.png",
      "uri": {
        "value": "public://2025-10/my-image.png",
        "url": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/2025-10/my-image.png"
      },
      "filemime": "image/png",
      "filesize": 1136628,
      "status": true,
      "created": "2025-10-16T18:27:37+00:00",
      "changed": "2025-10-16T18:28:31+00:00",
      "breadcrumbs": [],
      "links": {
        "1_1_square_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/1_1_square_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "240",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "2_1_large": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_1_large/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "512",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "2_1_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_1_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "240",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "2_3_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_3_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "480",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "3_2_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/3_2_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "320",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "7_2_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/7_2_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "300",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "full_content_width": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/full_content_width/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "2024",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "large": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/large/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "480",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "original": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/original/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "2024",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "self": {
          "href": "http://va-gov-cms.ddev.site/jsonapi/file/file/44ac8309-f23e-4286-85ce-f4958314e1bb"
        },
        "viewport_width": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/viewport_width/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "2024",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        }
      },
      "resourceIdObjMeta": {
        "alt": "Rating sandwiches",
        "title": null,
        "width": 840,
        "height": 2024,
        "drupal_internal__target_id": 53380
      },
      "uid": {
        "type": "user--user",
        "id": "7c46746a-7527-4403-abca-1c115643037a",
        "resourceIdObjMeta": {
          "drupal_internal__target_id": 5409
        }
      },
      "relationshipNames": ["uid"]
    },
    "image": {
      "type": "file--file",
      "id": "44ac8309-f23e-4286-85ce-f4958314e1bb",
      "drupal_internal__fid": 53380,
      "langcode": "en",
      "filename": "my-image.png",
      "uri": {
        "value": "public://2025-10/my-image.png",
        "url": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/2025-10/my-image.png"
      },
      "filemime": "image/png",
      "filesize": 1136628,
      "status": true,
      "created": "2025-10-16T18:27:37+00:00",
      "changed": "2025-10-16T18:28:31+00:00",
      "breadcrumbs": [],
      "links": {
        "1_1_square_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/1_1_square_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "240",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "2_1_large": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_1_large/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "512",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "2_1_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_1_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "240",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "2_3_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_3_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "480",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "3_2_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/3_2_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "320",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "7_2_medium_thumbnail": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/7_2_medium_thumbnail/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "300",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "full_content_width": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/full_content_width/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "2024",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "large": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/large/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "480",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "original": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/original/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "2024",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        },
        "self": {
          "href": "http://va-gov-cms.ddev.site/jsonapi/file/file/44ac8309-f23e-4286-85ce-f4958314e1bb"
        },
        "viewport_width": {
          "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/viewport_width/public/2025-10/my-image.png",
          "meta": {
            "linkParams": {
              "width": "2024",
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
              ]
            },
            "rel": [
              "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative"
            ]
          }
        }
      },
      "resourceIdObjMeta": {
        "alt": "Rating sandwiches",
        "title": "",
        "width": 840,
        "height": 2024,
        "drupal_internal__target_id": 53380
      },
      "uid": {
        "type": "user--user",
        "id": "7c46746a-7527-4403-abca-1c115643037a",
        "resourceIdObjMeta": {
          "drupal_internal__target_id": 5409
        }
      },
      "relationshipNames": ["uid"]
    },
    "relationshipNames": ["bundle", "uid", "thumbnail", "image"]
  },
  "relationshipNames": ["paragraph_type", "field_media"]
}
