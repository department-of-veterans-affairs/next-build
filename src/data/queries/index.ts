import { createQueries } from 'next-drupal-query'
import * as NewsStory from './newsStory'
import * as NewsStoryTeaser from './newsStoryTeaser'
import * as StoryListing from './storyListing'
import * as QuestionAnswer from './questionAnswer'
import * as ExpandableText from './expandableText'
import * as LinkTeaser from './linkTeaser'
import * as MediaImage from './mediaImage'
import * as Banner from './banners'
import * as PersonProfile from './personProfile'
import * as Button from './button'
import * as AudienceTopics from './audienceTopics'
import * as Alert from './alert'
import * as EmailContact from './emailContact'
import * as BenefitsHub from './benefitsHub'
import * as Wysiwyg from './wysiwyg'
import * as StaticPathResources from './staticPathResources'
import * as HeaderFooter from './headerFooter'

export const queries = createQueries({
  // standard Drupal entity data queries
  'node--news_story': NewsStory,
  'node--news_story--teaser': NewsStoryTeaser,
  'node--story_listing': StoryListing,
  'node--q_a': QuestionAnswer,
  'node--person_profile': PersonProfile,
  'node--landing_page': BenefitsHub, // "Benefits Hub Landing Page"
  'paragraph--audience_topics': AudienceTopics,
  'paragraph--button': Button,
  'paragraph--email_contact': EmailContact,
  'paragraph--expandable_text': ExpandableText,
  'paragraph--link_teaser': LinkTeaser,
  'paragraph--rich_text_char_limit_1000': Wysiwyg,
  'paragraph--wysiwyg': Wysiwyg,
  'media--image': MediaImage,
  'block--alert': Alert,

  // Custom queries 
  'banner--alerts_lookup': Banner,
  'header-footer-data': HeaderFooter,

  // Static Path Generation
  'static-path-resources': StaticPathResources,
})
