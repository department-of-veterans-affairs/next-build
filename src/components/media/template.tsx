import { Media as FormattedMedia } from './formatted-type'

export const Media = ({
  entityId,
  allowClicksOnThisImage,
  image,
}: FormattedMedia) => (
  <div
    data-template="paragraphs/media"
    data-entity-id={entityId}
    className="vads-u-display--block"
  >
    <div className="va-c-position--relative vads-u-display--inline-block vads-u-margin-y--1p5">
      {allowClicksOnThisImage && (
        <a
          aria-label="Open image in new tab"
          className="vads-u-margin-right--1p5 vads-u-margin-top--1p5 vads-u-text-decoration--none vads-u-display--flex vads-u-align-items--center expand-image-button va-c-position--absolute va-c-position-top-right-corner vads-u-justify-content--center"
          href={image.url}
          target="_blank"
        >
          <va-icon icon="zoom_out_map" size="3"></va-icon>
        </a>
      )}
      <img src={image.url} alt={image.alt} title={image.title} />
    </div>
  </div>
)
