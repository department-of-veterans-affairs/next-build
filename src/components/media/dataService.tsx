import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { EntityMetaInfo } from '@/lib/delegators/entityMetaProvider'
import { MediaImage, MediaResourceType } from '@/types/media'
import { MediaImageProps, MediaImageComponent } from '@/components/media'
import { absoluteURL } from '@/utils/helpers'

export const mediaImageDataService = function (
  entity: MediaImage,
  imageStyle
): MediaImageProps | null {
  if (!entity || !entity.image) return null

  return {
    url: absoluteURL(entity.image.uri?.url),
    styles: entity.image.links || imageStyle,
    alt: entity.image.resourceIdObjMeta?.alt,
    title: entity.image.resourceIdObjMeta?.title,
    width: entity.image.resourceIdObjMeta?.width,
    height: entity.image.resourceIdObjMeta?.height,
  }
}

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams()
  .addInclude(['image', 'image.thumbnail'])
  .addPageLimit(10)

/** Export information necessary to identify the component and query it.
 * See {@link EntityMetaInfo}
 */
export const Meta: EntityMetaInfo = {
  resource: MediaResourceType.Image,
  component: MediaImageComponent,
  dataService: mediaImageDataService,
  params: params,
}
