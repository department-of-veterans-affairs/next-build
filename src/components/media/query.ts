import { QueryFormatter, QueryParams } from 'next-drupal-query'
import { ParagraphMedia } from '@/types/drupal/paragraph'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { getNestedIncludes } from '@/lib/utils/queries'
import { Media } from './formatted-type'
// import { queries } from '@/lib/drupal/queries'

export const params: QueryParams<null> = () =>
  new DrupalJsonApiParams().addInclude(
    getNestedIncludes('field_media', 'media--image')
  )

export const formatter: QueryFormatter<ParagraphMedia, Media | null> = (
  entity: ParagraphMedia
) => {
  if (!entity.field_media || !entity.field_media.image) return null
  return {
    type: entity.type as ParagraphMedia['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id ?? null,
    // image: queries.formatData('media--image', entity.field_media),
    image: {
      url: entity.field_media.image.uri.url,
      alt: entity.field_media.image.resourceIdObjMeta.alt,
      title: entity.field_media.image.resourceIdObjMeta.title,
    },
    allowClicksOnThisImage: entity.field_allow_clicks_on_this_image,
  }
}
