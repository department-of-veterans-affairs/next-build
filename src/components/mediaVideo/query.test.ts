/**
 * @jest-environment node
 */

import { DrupalMediaVideo } from '@/types/drupal/media'
import { queries } from '@/data/queries/'
import mockData from '@/components/mediaImage/mock.json'

//eslint-disable-next-line
const mediaVideoMock: DrupalMediaVideo | any = mockData

describe('Media image returns formatted data', () => {
  test('outputs formatted data', () => {
    const formattedData = mediaVideoMock
    expect(queries.formatData('media--video', formattedData)).toMatchSnapshot()
  })
})
