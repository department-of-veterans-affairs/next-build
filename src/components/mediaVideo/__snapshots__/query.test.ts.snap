// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Media image returns formatted data outputs formatted data 1`] = `
{
  "field_description": null,
  "field_duration": undefined,
  "field_media_video_embed_field": undefined,
  "id": "7252cad5-7ff8-49d3-add4-7f5b009761ce",
  "name": "health-care.png",
  "type": "media--image",
}
`;
